.github/workflows/sdk-build.yml

```diff
@@ -34,6 +34,8 @@ jobs:
         run: pydoclint --exclude='.*/build/.*' src
       - name: Lint with pylint
         run: pylint src --disable=all --enable=C0103 --ignore=build
+      - name: Check and add license headers
+        run: python scripts/license_header.py src
```

Comment:
I like the addition of a Python script for adding license headers 😋 
However, I'm wondering if this could be optimized a bit more for CI?

* Rather than making edits to the files in CI, a check for whether the script is being run in CI could save execution time. If in a CI environment, we can skip making the edits since those changes won't be pushed up anyway. As for some ideas of detecting CI, we could check for the environment variable `CI`, or the script could be updated to support a `dry-run` option (that does not make any changes).
* If in CI, the print statement "Adding license header to: " could be refactored so that it just prints the individual file paths that are missing a license header.